src/functional/fencil_processors/source_modules/source_modules.py

```diff
@@ -15,11 +15,37 @@
 
 
 from dataclasses import dataclass
-from typing import Sequence
+from typing import Generic, Protocol, TypeVar
 
 import numpy
 
 
+class SupportedLanguageProtocol(Protocol):
+    """Ensures consistent code formatting along the pipeline."""
+
+    @property
+    def name(self) -> str:
+        ...
+
+    def format_source(self, source_code: str) -> str:
+        ...
+
+
+class IncludeImplementationLanguageProtocol(SupportedLanguageProtocol, Protocol):
```

Comment:
How about `LanguageFileExtensionsProtocol` or `SourceFileExtensionsProtocol` or something similar? I interpreted the name as "include the ImplementationLanguageProtocol" and it was pretty confusing.

Reply:
* `LanguageFileExtensionsProtocol`: looks like `Python` might want to implement that. Except python does not split "include" and "implementation" files. -> Too general
* `SourceFileExtensionsProtocol`: `Python` also has source file extensions...
* `LanguagesWhichSplitImplementationAndIncludeFilesProtocol`: too long
* `HeaderAndImplementationFilesLanguageProtocol`: a bit long.

Considering we don't really need the `Protocol`, maybe `LanguageWithHeaders`?

Reply:
A bit awkward still, but I can live with that.